wtransport/src/tls.rs

```diff
@@ -672,28 +673,60 @@ pub mod client {
     ///
     /// Therefore, it's advisable to use it judiciously, and avoid using it in
     /// production environments.
-    #[derive(Default)]
-    pub struct NoServerVerification {}
+    #[derive(Debug)]
+    pub struct NoServerVerification {
+        supported: WebPkiSupportedAlgorithms,
+    }
+
+    impl Default for NoServerVerification {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
 
     impl NoServerVerification {
         /// Creates a new instance of `NoServerVerification`.
         pub fn new() -> NoServerVerification {
-            NoServerVerification {}
+            NoServerVerification {
+                supported: rustls::crypto::ring::default_provider()
+                    .signature_verification_algorithms,
+            }
         }
     }
 
     impl ServerCertVerifier for NoServerVerification {
         fn verify_server_cert(
             &self,
-            _end_entity: &rustls::Certificate,
-            _intermediates: &[rustls::Certificate],
-            _server_name: &rustls::ServerName,
-            _scts: &mut dyn Iterator<Item = &[u8]>,
+            _end_entity: &rustls_pki_types::CertificateDer,
+            _intermediates: &[rustls_pki_types::CertificateDer],
+            _server_name: &rustls_pki_types::ServerName,
             _ocsp_response: &[u8],
-            _now: std::time::SystemTime,
+            _now: rustls_pki_types::UnixTime,
         ) -> Result<ServerCertVerified, rustls::Error> {
             Ok(ServerCertVerified::assertion())
         }
+
+        fn verify_tls12_signature(
+            &self,
+            message: &[u8],
+            cert: &CertificateDer<'_>,
+            dss: &rustls::DigitallySignedStruct,
+        ) -> Result<rustls::client::danger::HandshakeSignatureValid, rustls::Error> {
+            rustls::crypto::verify_tls12_signature(message, cert, dss, &self.supported)
```

Comment:
I am wondering the downside to just assert the signature

Reply:
That would seemingly violate [the documentation](https://docs.rs/rustls/latest/rustls/client/danger/trait.ServerCertVerifier.html#tymethod.verify_tls12_signature):
> If and only if the signature is valid, return Ok(HandshakeSignatureValid). Otherwise, return an error – rustls will send an alert and abort the connection.

Also, FWIW, this matches the old behavior (which was [provided by a default fn in the trait](https://docs.rs/rustls/0.21.8/src/rustls/verify.rs.html#151)).

Reply:
Nice thanks!